### PR TITLE
chore(deps): Update dependencies

### PR DESCRIPTION
**Pull request by reneleonhardt**
### Updates
* Kotlin 1.8.22 to 1.9.24
* android-plugin 8.1.1 to 8.4.1
* classgraph 4.8.170 to 4.8.172
* compile-testing 0.3.2 to 0.4.1
* detekt 1.23.0 to 1.23.6
* dokka 1.8.20 to 1.9.20
* federation 4.4.1 to 5.0.0
* graalvm 0.10.1 to 0.10.2
* Gradle Wrapper (latest, no version shown)
* icu 74.2 to 75.2
* jackson 2.15.4 to 2.17.1
* kotlinx-benchmark 0.4.8 to 0.4.10
* kotlinx-coroutines 1.7.3 to 1.8.1
* kotlinx-serialization 1.5.1 to 1.6.3 **(3 test cases failing in GenerateKotlinxClientIT)**
* ktlint-core 0.45.2 to 1.2.1 **(many rules had to be disabled, reformat required)**
* ktlint-plugin 10.3.0 to 12.1.1
* ktor 2.3.10 to 2.3.11
* logback 1.5.4 to 1.5.6
* maven-plugin-annotation 3.12.0 to 3.13.0
* mockk 1.13.7 to 1.3.11
* nexus-publish-plugin 1.3.0 to 2.0.0
* poet 1.14.2 to 1.17.0
* reactor-core 3.5.15 to 3.6.6
* slf4j 2.0.12 to 2.0.13
* spring 6.0.18 to 6.1.8
* spring-boot 3.1.10 to 3.3.0
* wiremock 3.5.2 to 3.6.0

### Added
* spring-context (was needed for spring-web in graphql-kotlin-spring-client)

### Improvements
* Synchronized all covered ratio values to current values
* Let Dependabot update GitHub Actions

### TODO
* Fix GraphQLTestUtils and GenerateKotlinxClientIT after Kotlin 1.9 upgrade
* Remove disabled rules in .editorconfig, then `./gradlew ktlintFormat`

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 version: 2
 updates:
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
```
Comment (dariuszkuc) [inline .github/dependabot.yml:6]: why do we need to run it daily? gradle one is scheduled on weekly basis

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -1,53 +1,46 @@
...
+dokka = "1.9.20"
 jacoco = "0.8.12"
-# klint gradle plugin breaks with 0.46.x+
-ktlint-core = "0.45.2"
-ktlint-plugin = "10.3.0"
+ktlint-core = "1.2.1"
+ktlint-plugin = "12.1.1"
 maven-plugin-development = "0.4.3"
-nexus-publish-plugin = "1.3.0"
+nexus-publish-plugin = "2.0.0"
```
Comment (dariuszkuc) [inline gradle/libs.versions.toml:43]: I believe this change requires change to release workflow -> i.e. task was renamed from `closeAndReleaseSonatypeStagingRepository` to `closeAndReleaseStagingRepositories`

File: .github/dependabot.yml (hunk 1 above)
Comment (reneleonhardt) [reply .github/dependabot.yml]: Oversight, changed it back to weekly

File: gradle/libs.versions.toml (hunk 2 above)
Comment (reneleonhardt) [reply gradle/libs.versions.toml:43]: Your workflow uses `closeAndReleaseSonatypeStagingRepository` and succeeded, the breaking change affected only `closeAndReleaseStagingRepository`:
https://github.com/gradle-nexus/publish-plugin/releases/tag/v2.0.0

**Comment by dariuszkuc** [comment]
nit: we probably don't need to pin the coverage values to the exact ones, i.e. we should have some wiggle room when making some trivial changes

File: clients/graphql-kotlin-spring-client/build.gradle.kts [hunk 3]
```diff
@@ -10,6 +10,7 @@ dependencies {
     api(projects.graphqlKotlinClientJackson)
     api(libs.kotlinx.coroutines.reactor)
     api(libs.spring.webflux)
+    api(libs.spring.context)
```
Comment (dariuszkuc) [inline clients/graphql-kotlin-spring-client/build.gradle.kts:13]: why do we need explicit spring context dependency here?

**Review by dariuszkuc — APPROVED** [review]

File: clients/graphql-kotlin-spring-client/build.gradle.kts (hunk 3 above)
Comment (reneleonhardt) [reply clients/graphql-kotlin-spring-client/build.gradle.kts:13]: Spring changed their dependency management I guess 🤷

`:graphql-kotlin-spring-client:compileTestKotlin`

> Supertypes of the following classes cannot be resolved. Please make sure you have the required dependencies in the classpath:
>     class org.springframework.http.client.reactive.ReactorClientHttpConnector, unresolved supertypes: org.springframework.context.SmartLifecycle
>

**Comment by reneleonhardt** [comment]
> nit: we probably don't need to pin the coverage values to the exact ones, i.e. we should have some wiggle room when making some trivial changes

I reverted the new values except where the difference was too big.

**Pull request merged**